Automattic/pocket-casts-ios
# Tracks: Adds analytics to the Auto Download settings

| 📘 Project: #154 | 🛫 Depends on: #309 |
|:---:|:---:|

- `settings_auto_download_shown`: When the Auto Download setting is shown to the user
- `settings_auto_download_up_next_toggled`: When the user toggles the Auto Download Up next option
- `settings_auto_download_new_episodes_toggled`: When the user toggles the Auto Download New Episodes option
- `settings_auto_download_podcasts_changed`: When the use changes which podcasts should be auto downloaded
- `settings_auto_download_filters_changed`: When the use changes which filters should be auto downloaded
- `settings_auto_download_only_on_wifi_toggled`: When the user toggles the Only On WiFi option

## To test

1. Go to the profile tab
2. Tap on the cog
3. Tap on Auto Download
4. ✅ `🔵 Tracked: settings_auto_download_shown`
9. Go through each setting and change its value
12. Check console and verify each setting has a corresponding event in console and the value displayed is the value that you selected
13. If the option changed is a switch then the event should be suffixed `_toggle` if it's a value change it should be suffixed with `_changed`


## Checklist
